File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.ts
function frequencySort(s: string): string {
  const freq = new Map<string, number>();
  for (const char of s) {
    freq.set(char, (freq.get(char) ?? 0) + 1);
  }

  const list: [string, number][] = Array.from(freq.entries());
  list.sort((a, b) => b[1] - a[1]);

  let result = "";

  list.forEach((val) => {
    for (let i = 0; i < val[1]; i++) {
      result += val[0];
    }
  });

  return result;
}

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.swift
class Solution {
    func frequencySort(_ s: String) -> String {
        var freqs: Dictionary<Character, Int> = [:]
        
        s.forEach({ it in
            if let freq = freqs[it] {
                freqs[it] = freq + 1
            } else {
                freqs[it] = 1
            }
        })
        
        let sorted =  freqs.sorted(by: { (p1, p2) in
            return p1.value > p2.value
        })
        
        var result = ""
        sorted.forEach({ pair in
            for _ in 0 ..< pair.value {
                result += String(pair.key)
            }
        })
        return result
    }
}
